pkg/storage/shared_storage.go

```diff
@@ -48,6 +48,9 @@ func (r *externalStorageReader) ReadAt(ctx context.Context, p []byte, offset int
 	for n := 0; n < len(p); {
 		nn, err := reader.Read(ctx, p[n:])
 		if err != nil {
+			if errors.Is(err, io.EOF) {
```

Comment:
So we don't return any error, but we didn't populate the entire buffer. How is the caller supposed to know which part of the buffer actually has correct data?